tdms/src/fields/split.cpp

```diff
@@ -0,0 +1,38 @@
+#include "numeric.h"
+#include "field.h"
+
+using namespace std;
+
+
+void SplitField::allocate() {
+
+  for (auto component : {xy, xz, yx, yz, zx, zy}){
```

Comment:
note for reviewers: this is wrong – it should be an iteration over references and has been fixed in a later commit